{
  try {
    listener.onComplete(result);
  }
 catch (  Throwable x) {
    LOG.info("Exception while notifying listener " + listener,x);
  }
}
